annotation/annotation-compatibility-stub/build.gradle

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import androidx.build.AndroidXComposePlugin
+import androidx.build.JetbrainsAndroidXPlugin
+import androidx.build.LibraryType
+import org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType
+import org.jetbrains.kotlin.gradle.tasks.KotlinNativeCompile
+
+plugins {
+    id("AndroidXPlugin")
+
+    //  TODO move all functionality to Compose-independent plugin (`collection` shouldn't depend on Compose concepts)
+    id("AndroidXComposePlugin")
+    id("JetbrainsAndroidXPlugin")
+}
+
+AndroidXComposePlugin.applyAndConfigureKotlinPlugin(project)
+JetbrainsAndroidXPlugin.applyAndConfigure(project)
+
+androidXComposeMultiplatform {
+    js()
+    wasm()
+    darwin()
+
+    linuxX64()
+    linuxArm64()
+}
+
+kotlin {
+    jvm()
+    // Not all modules can have these targets, so declare them here instead of androidXComposeMultiplatform.
+    // We can support the kotlin-native targets supported by kotlin coroutines:
+    // https://github.com/Kotlin/kotlinx.coroutines/blob/master/gradle/compile-native-multiplatform.gradle
+    // NOTE: Keep this list in sync with runtime-saveable's targets.
+    watchosArm64()
+    watchosArm32()
+    watchosX64()
+    watchosSimulatorArm64()
+    tvosArm64()
+    tvosX64()
+    tvosSimulatorArm64()
+    mingwX64()
+
+    sourceSets {
+        commonMain {
+            dependencies {
+                api("androidx.annotation:annotation:1.9.1")
```

Comment:
Having the dependency here will require to re-publish the stub for any androidx library update

Reply:
We'll move the dependency to androidx to the modules where it's needed.
So the stub will be published only once.

Reply:
The dependency in a stub is needed. Otherwise there is no link from the stub to the new lib.

Reply:
yes, it's needed now if somone depends directly on our org.jetbrains.anotation-internal (stubbed klib), they would be able to use its APIs.

Reply:
> depends directly on our org.jetbrains.anotation-internal

not only directly, but indirectly, for example, on `lifecycle`